### ISSUE TITLE
compileGroovy ends up removing compileJava output from build/classes

### ISSUE THREAD

**Issue by kristofdepypere** [gradle/gradle#917]
I have a multi-project build and one of the subprojects has the java and groovy plugin applied. The groovy source code depends on some java classes of the same subproject. However when performing compileGroovy it seems that all the classes that were previously compiled by compileJava are removed from build/classes and groovy complains that it cannot find the classes.

When performing only the compileJava task, all classes are properly compiled and outputted in the build/classes folder of my subproject.

### Expected Behavior

Expected is that compileGroovy can safely rely on the previous output of the compileJava task.

### Current Behavior

All output of the compileJava task seems to have been removed.

### Context

Previously my subproject compiled all using the compileGroovy task: 

```gradle
sourceSets {
	main {
		java {
			srcDirs = []
		}
		groovy {
			srcDirs += ["src/main/java"]
		}
	}
}
```

I wanted to get rid of this and make sure groovy relies on java and not vice versa. I changed my code and removed this block, but then I bump into the problem described.
### Your Environment

 * Build scan URL: https://scans.gradle.com/s/isi4ren6g3ndu#infrastructure
The scan above is from when everything was still compiled using compileGroovy only.

**Comment by wolfs** [comment]
In order to fix your build you need to delete your `.gradle` directory in your project and run a clean. The problem which you are facing is caused by the following:
The `GroovyCompile` task snapshots its outputs after each run. So it remembers that it produced all the `.class` files for the Java files. Now the Groovy compiler is actually a `CleaningGroovyCompiler` which removes all the outputs of the previous execution before it starts to compile again. This yields to the behavior you are seeing. The history for the outputs is persisted in the `.gradle` directory within your project directory. So cleaning that removes the "memory" of the `GroovyCompile` task.

**Comment by kristofdepypere** [comment]
k, thx. That worked. I assume after my change all developers have to do exactly the same in order to avoid compilation problems ?

**Comment by eriwen** [comment]
@kristofdepypere Thank you for the well-formed issue. We may or may not be able to address this as we fix #821 which is "in progress".

**Comment by oehme** [comment]
This has been fixed as part of the work on #821
